hazelcast/src/main/java/com/hazelcast/cache/impl/AbstractCacheService.java

```diff
@@ -869,4 +873,35 @@ public void onClusterStateChange(ClusterState newState) {
             expManager.onClusterStateChange(newState);
         }
     }
+
+    @Override
+    public void provideDynamicMetrics(MetricDescriptor descriptor, MetricsCollectionContext context) {
+        Map<String, LocalCacheStats> stats = getStats();
```

Comment:
(maybe not scope of this PR)
what if stats calculation is disabled, does metrics show that?

Reply:
Addressed in a separate commit. Needed to add ifs for every data structure to respect `setStatisticsEnabled(false)`.

Checked with @puzpuzpuz and with MC in local, we're safe to exclude the statistics if it's disabled.